.github/workflows/test_python.yml

```diff
@@ -0,0 +1,11 @@
+name: tests
+on: [pull_request, push]
+jobs:
+  test_python:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: pip install -r requirements-dev.txt
+      - run: flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+      - run: pip install -r requirements.txt
+      - run: pytest
```

Comment:
I think the steps for running `black`, `mypy` and `isort` should be included with `|| true` and also `pytest --doctest-modules .` if there are any doctests.

Reply:
@dhruvmanila, thanks for the review, `black || true` and `isort || true` don't test anything about the code that won't change without considerable work. The core files are not black compliant, and the imports are not sorted correctly, and have other unspecified issues that will need to be fixed. You could add more Flake8 checks as these get tidied up.

The examples in the docstrings don't have any assertions, and tend to make live calls to Open Library. I don't think these should be run at all. I have added some pytest protection  (`pytest-socket`) to prevent live calls, and fixed the bot unit tests which made many. To test the API calls properly Mocks are needed, and I don't think docstrings are the best place for those. The current docstrings appear to be designed as usage examples, not tests. 

Additionally the Abstract Bot class hijacks arguments passed to pytest, so I'm not sure how you would run the doctests cleanly anyway.  I've added unittest code to enable arguments for pytest, e.g. `-s` for debugging locally.  I'm not suggesting we want doctests here at all. The unit tests cover the complex calls that need mocking sufficiently.

I did enable mypy type checking per your comment. The previous `mypy || true` wasn't testing anything as there were simple issues that had gone undetected, now resolved in this PR.

`|| true` is not a good practice for testing, and especially not for CI where you want fast meaningful feedback.